applications/launchpad/gui-react/src/pages/onboarding/index.tsx

```diff
@@ -1,61 +1,60 @@
 import { useState, useEffect } from 'react'
 import TBotPrompt from '../../components/TBot/TBotPrompt'
+import { OnboardingMessagesMap } from '../../config/onboardingMessagesConfig'
 
 const Onboarding = ({ close }: { close: () => void }) => {
   const [trigger, fireTrigger] = useState(false)
 
-  const [messages, setMessages] = useState([
-    'first message',
-    'second message',
-    {
-      content: 'need to wait longer...',
-      wait: 6000,
-    },
-    <div key='key1'>
-      <p>Push new message</p>
-      <button onClick={() => fireTrigger(true)}>Push</button>
-    </div>,
-  ])
+  const [messages, setMessages] = useState(OnboardingMessagesMap)
+
+  // Example of trigger to push new messages
+  //   <div key='key1'>
+  //     <p>Push new message</p>
+  //     <button onClick={() => fireTrigger(true)}>Push</button>
+  //   </div>,
+  //
 
   // If we use currentIndex = 1, it will not show loading dots before rendering first message
-  const [current, setCurrent] = useState(1)
+  const [current, setCurrent] = useState(0)
 
   useEffect(() => {
     if (trigger) {
       const newMsgs = messages.slice()
-      newMsgs.push('newly pushed')
+      newMsgs.push({ content: 'Newly pushed', barFill: 0 })
       setMessages(newMsgs)
       setCurrent(newMsgs.length - 1)
-      // setCurrent(newMsgs.length) // it won't show loading dots when currentIndex is equal to the messages length
+      setCurrent(newMsgs.length) // it won't show loading dots when currentIndex is equal to the messages length
```

Comment:
you can avoid adding a comment with a named function, like
`const hideLoadingDots = () => setCurrent(newMsgs.length)`

Reply:
And this gets called on the next line, or made into an IIFE?